starter-code/src/QueueDataStructure.js

```diff
@@ -1,2 +1,40 @@
 function QueueDataStructure () {
+    this.queueControl =[];
+    this.MAX_SIZE=1;
 }
+
+QueueDataStructure.prototype.isEmpty=function(){
+
+  if (this.queueControl.length===0){
+    return true;
+  } else {
+    return false;
+  }
+};
+QueueDataStructure.prototype.canEnqueue=function(){
+if (this.MAX_SIZE===this.queueControl.length){
+  return false;
+} else {
+  this.dequeue();
```

Comment:
this method should purely return true or false, and shouldn't actually change the queue, so this line should not be here